doc/adr/0001-reactive-reassignment.md

```diff
@@ -0,0 +1,49 @@
+# 1. Reactive reassignment
+
+Date: 2018-11-21
+
+## Status
+
+Accepted
+
+## Context
+
+In current architecture, we receive always a batch of products. For this batch, we check if any products need reassignment.
+In order to find out, we search for conflicting products with query by SKUs from variants and slugs from all locale and both current and staged.
+This query can get too big for GET request and cannot be used.  
+
+## Decision
+
+Variant reassignment will be called only in case of errors and it will solve only the failed product.
+
+## Consequences
+Refactoring needs to be done both in reassignment module and also modules that uses reassignment (right now only sphere-product-import).
```

Comment:
"also modules that use~s~"

Reply:
https://github.com/commercetools/commercetools-node-variant-reassignment/pull/77/commits/6b966f6aee1249d5398b4b636b6c9803f78e56ef#diff-69de50b194fdf22d40e38b82f2045440R19